src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ClosureConversion.cs

```diff
@@ -1630,6 +1636,39 @@ private BoundNode RewriteLambdaConversion(BoundLambda node)
             }
 
             return result;
+
+            bool HasTypeArgumentsFromReferencedMethod(TypeSymbol symbolToCheck)
```

Comment:
>HasTypeArgumentsFromReferencedMethod [](start = 17, length = 36)

Type checks for specific symbol types in this function look suspicious. It would be good to add a comment explaining what exactly we are trying to accomplish here.

Reply:
Will do that.

Also it would be nice if somebody review the function very closely. It looks too complicated for me but I haven't found an easier way to check the type.

Reply:
> Also it would be nice if somebody review the function very closely. It looks too complicated for me but I haven't found an easier way to check the type.

We will be able to do that once we know what this function is supposed to do

---
In reply to: [436855315](https://github.com/dotnet/roslyn/pull/44939#discussion_r436855315) [](ancestors = 436855315)

Reply:
Add a comment describing the method.

Reply:
> Added a comment describing the method.

It looks like you simply translated C# inside the method to English. What is the referenced method? What is the symbol to check? What kind of dependency between them we are trying to detect?

---
In reply to: [437689304](https://github.com/dotnet/roslyn/pull/44939#discussion_r437689304) [](ancestors = 437689304)

Reply:
Not actual anymore since I've deleted the method.